src/filter/mod.rs

```diff
@@ -189,6 +188,16 @@ impl Builder {
         builder
     }
 
+    /// Insert the directive replacing any directive with the same name.
+    fn insert_directive(&mut self, mut directive: Directive) {
+        let by_name = |d: &Directive| d.name == directive.name;
+        if let Some(pos) = self.directives.iter().position(by_name) {
```

Comment:
Any particular reason why you didn't write that as

```suggestion
        if let Some(pos) = self.directives.iter().position(|d: &Directive| d.name == directive.name) {
```

?

Reply:
Hi, sorry for the slow reply. Just catching up after holiday + illness. I believe I chose to do this due to the way rustfmt splits up the long `if let` line:

```suggestion
        if let Some(pos) = self
            .directives
            .iter()
            .position(|d: &Directive| d.name == directive.name)
        {
```

Though I'm happy with either suggestion. Either way I'll rebase and add to the commit comment.